### ISSUE TITLE
ASR model with unidirectional lstm and Monotonic attention not converging

### ISSUE THREAD

**Issue by manish-kumar-garg** [tensorflow/lingvo#191]
I made some changes to asr model.
encoder -> unidirectional lstm and monotonic attention.
After around 10000 steps, I can see that the loss is around 0.1

```
I0106 06:35:36.474763 140652338456320 summary_utils.py:341] Steps/second: 0.068993, Examples/second: 7.080114
I0106 06:35:36.476172 140652338456320 trainer.py:520] step: 10325, steps/sec: 0.07, examples/sec: 7.08 fraction_of_correct_next_step_preds:0.9612695 fraction_of_correct_next_step_preds/logits:0.9612695 grad_norm/all/loss:0.45247629 grad_scale_all/loss:1 has_nan_or_inf/loss:0 log_pplx:0.12303153 log_pplx/logits:0.12303153 loss:0.12303153 loss/logits:0.12303153 num_samples_in_batch:96 token_normed_prob:0.88425374 token_normed_prob/logits:0.88425374 var_norm/all/loss:401.54611
```

**But WER is around 0.8**

Can you suggest something, why is this happening? What parameters should I try to change?

**Comment by manish-kumar-garg** [comment]
@jonathanasdf  Your comments can be helpful!
Should I decrease the learning rate?

**Comment by jonathanasdf** [comment]
I'm trying to ask the domain experts but haven't gotten a reply yet.

From the training logs 96% correct next step preds the model should be training fine, so there might be some kind of data mismatch?

What is WER for decoder_train? If that is low the model can be severely overfitting but I find that unlikely.

**Comment by manish-kumar-garg** [comment]
@jonathanasdf After some more training:

Around 20K steps: wer ~ 0.40
while after noise addition, model seems to be not converging.
At 40K steps: wer ~ 0.97
At 48K steps: wer ~ 0.97

```
I0110 10:24:21.001313 139618548692736 trainer.py:519] step: 19996, steps/sec: 0.15, examples/sec: 16.50 fraction_of_correct_next_step_preds:0.97068101 fraction_of_correct_next_step_preds/logits:0.97068101 grad_norm/all/loss:0.28118196 grad_scale_all/loss:1 has_nan_or_inf/loss:0 log_pplx:0.093538016 log_pplx/logits:0.093538016 loss:0.093538016 loss/logits:0.093538016 num_samples_in_batch:96 token_normed_prob:0.91070336 token_normed_prob/logits:0.91070336 var_norm/all/loss:475.66272
.
.
I0109 06:10:43.689600 140595444033280 trainer.py:519] step: 34999, steps/sec: 0.10, examples/sec: 10.41 fraction_of_correct_next_step_preds:0.62987983 fraction_of_correct_next_step_preds/logits:0.62987983 grad_norm/all/loss:0.060645018 grad_scale_all/loss:1 has_nan_or_inf/loss:0 log_pplx:1.1489006 log_pplx/logits:1.1489006 loss:1.1489006 loss/logits:1.1489006 num_samples_in_batch:96 token_normed_prob:0.31698507 token_normed_prob/logits:0.31698507 var_norm/all/loss:303.03281
.
.
I0110 10:20:36.684551 140595444033280 trainer.py:519] step: 49133, steps/sec: 0.14, examples/sec: 14.05 fraction_of_correct_next_step_preds:0.64321792 fraction_of_correct_next_step_preds/logits:0.64321792 grad_norm/all/loss:0.063578233 grad_scale_all/loss:1 has_nan_or_inf/loss:0 log_pplx:1.0926927 log_pplx/logits:1.0926927 loss:1.0926927 loss/logits:1.0926927 num_samples_in_batch:96 token_normed_prob:0.33531237 token_normed_prob/logits:0.33531237 var_norm/all/loss:328.36472

```
Is this expected behavior? Should I wait more? 
What parameters can I change and try?

**Comment by manish-kumar-garg** [comment]
@jonathanasdf @drpngx 
 Do you think, changing the noise addition step from 20K to more maybe 30K will help?
